modules/ui/fields/date.js

```diff
@@ -0,0 +1,268 @@
+            month: format,
+            timeZone: 'UTC',
+        });
+        let parts = longFormat.formatToParts(new Date(Date.UTC(0, month, 1)));
+        let monthPart = parts.find(p => p.type === 'month');
+        return monthPart && monthPart.value;
+    }
+
+    let monthNames = Array.from({length: 12}, (_, i) => getMonthName(i, 'long'))
+        .filter(m => m);
+    let alternativeMonthNames = Array.from({length: 12}, (_, i) => {
+        return ['numeric', '2-digit', 'long', 'short', 'narrow']
+            .map(format => getMonthName(i, format))
+            .filter(m => m);
+    });
+
+    let monthCombo = uiCombobox(context, 'date-month')
+        .data(monthNames.map((monthName, i) => {
+            return {
+                id: i + 1,
+                value: monthName,
+                title: monthName,
+                display: selection => selection.append('span')
+                    .attr('class', 'localized-text')
+                    .text(monthName),
+                terms: alternativeMonthNames[i],
+            };
+        }));
+
+    function date(selection) {
+
+        var wrap = selection.selectAll('.form-field-input-wrap')
+            .data([0]);
+
+        wrap = wrap.enter()
+            .append('div')
+            .attr('class', 'form-field-input-wrap form-field-input-' + field.type)
+            .merge(wrap);
+
+        yearInput = wrap.selectAll('input.date-year')
+            .data([0]);
+        eraInput = wrap.selectAll('input.date-era')
+            .data([0]);
+        monthInput = wrap.selectAll('input.date-month')
+            .data([0]);
+        dayInput = wrap.selectAll('input.date-day')
+            .data([0]);
+
+        formatParts.forEach(part => {
+            switch (part) {
+                case 'year':
+                    yearInput = yearInput.enter()
+                        .append('input')
+                        .attr('type', 'number')
+                        .attr('class', 'date-year')
+                        .attr('id', field.domId)
+                        .call(utilNoAuto)
+                        .merge(yearInput);
+                    break;
+                case 'era':
+                    eraInput = eraInput.enter()
+                        .append('input')
+                        .attr('type', 'text')
+                        .attr('class', 'date-era')
+                        .call(eraCombo)
+                        .merge(eraInput);
+                    break;
+                case 'month':
+                    monthInput = monthInput.enter()
+                        .append('input')
+                        .attr('type', 'text')
+                        .attr('class', 'date-month')
+                        .call(monthCombo)
+                        .merge(monthInput);
+                    break;
+                case 'day':
+                    dayInput = dayInput.enter()
+                        .append('input')
+                        .attr('type', 'number')
```

Comment:
The year and day fields will need to be reworked a bit once we merge in the number formatting improvements in openstreetmap/iD#8769.